packages/@storefront/breadcrumbs/src/refinement-crumb/index.ts

```diff
@@ -1,3 +1,49 @@
-import { view } from '@storefront/core';
+import { provide, tag, Tag } from '@storefront/core';
 
-view('gb-refinement-crumb', require('./index.html'));
+@provide('refinementCrumb')
+@tag('gb-refinement-crumb', require('./index.html'))
+class RefinementCrumb {
+  init() {
+    this.updateState();
+  }
+
+  onUpdate() {
+    this.updateState();
+  }
+
+  updateState() {
+    let label;
+
+    if (this.props.range) {
+      label = this.props.low + ' - ' + this.props.high;
+    } else if (this.props.boolean) {
+      label = this.props.navigationLabel;
+    } else {
+      label = this.props.value;
+    }
+
+    this.state = {
+      ...this.props,
```

Comment:
Why are you spreading props into state?

Reply:
Reverted this changes and made it so that the component doesn't require state anymore to do the logic checks.